web/src/pages/playground.tsx

```diff
@@ -0,0 +1,253 @@
+            <SectionHeader text="Models"/>
+            <label for="completions-model" className="col-span-2">Completions model:</label>
+            <select name="completions-model" className="col-span-2" value={settings.completions} onChange={update('completions')}>
+                {COMPLETION_MODELS.map((name) => (
+                    <option value={name}>{name}</option>
+                ))}
+            </select>
+
+            <label for="encoder" className="col-span-2">Encoder:</label>
+            <select name="encoder" className="col-span-2" value={settings.encoder} onChange={update('encoder')}>
+                {ENCODERS.map((name) => (
+                    <option value={name}>{name}</option>
+                ))}
+            </select>
+
+            <SectionHeader text="Token options"/>
+            <NumberInput field="numTokens" label="Tokens" min="1"/>
+            <NumberInput field="tokensBuffer" label="Number of tokens to leave as a buffer when calculating remaining tokens" min="0" max={settings.tokensBuffer} />
+
+            <SectionHeader text="Prompt options"/>
+            <NumberInput field="topKBlocks" label="Number of blocks to use as citations" min="1"/>
+            <NumberInput field="maxHistory" label="The max number of previous interactions to use" min="0"/>
+
+            <Slider field="contextFraction" label="Approximate fraction of num_tokens to use for citations text before truncating"/>
+            <Slider field="historyFraction" label="Approximate fraction of num_tokens to use for history text before truncating"/>
+        </div>
+    )
+}
+
+type ChatPromptParams = {
+    settings: LLMSettings,
+    query: string,
+    history: Entry[],
+    updateSettings: (updater: (settings: LLMSettings) => LLMSettings) => void,
+}
+
+const ChatPrompts = ({settings, query, history, updateSettings}: ChatPromptParams) => {
+    const updatePrompt = (...path) => (event) => {
+        const newPrompts = {...updateIn(settings.prompts, path, event.target.value)};
+        updateSettings((settings) => ({...settings, prompts: newPrompts}))
+    };
+    return (
+        <div class="chat-prompts w-[400px] flex-none outline-black border-2 mx-5 p-5">
+            <details open>
+                <summary>Source prompt</summary>
+                <TextareaAutosize
+                    className="border border-gray w-full px-1"
+                    value={settings.prompts.source.prefix}
+                    onChange={updatePrompt('source', 'prefix')}
+                />
+                <div>(This is where sources will be injected)</div>
+                {history.length > 0 && (
+                <TextareaAutosize
+                    className="border border-gray w-full px-1"
+                    value={settings.prompts.source.suffix}
+                    onChange={updatePrompt('source', 'suffix')}
+                />
+                )}
+            </details>
+            {history.length > 0 && (
```

Comment:
This will display the text of the previous interactions, as the last `maxHistory` items will be used:
![Uploading image.png…]()